tripal/includes/TripalImporter.inc

```diff
@@ -0,0 +1,775 @@
+   */
+  private $num_handled;
+
+  /**
+   * The interval when the job progress should be updated. Updating the job
+   * progress incurrs a database write which takes time and if it occurs to
+   * frequently can slow down the loader.  This should be a value between
+   * 0 and 100 to indicate a percent interval (e.g. 1 means update the
+   * progress every time the num_handled increases by 1%).
+   */
+  private $interval;
+
+  /**
+   * Each time the job progress is updated this variable gets set.  It is
+   * used to calculate if the $interval has passed for the next update.
+   */
+  private $prev_update;
+
+  /**
+   * The job that this importer is associated with.  This is needed for
+   * updating the status of the job.
+   */
+  protected $job;
+
+  /**
+   * The drupal logger for tripal. This allows any of the importers to 
+   * send log messages.
+   */
+  protected $logger;
+
+  /**
+   * The arguments needed for the importer. This is a list of key/value
+   * pairs in an associative array.
+   */
+  protected $arguments;
+
+  /**
+   * The ID for this import record.
+   */
+  protected $import_id;
+
+  /**
+   * Prior to running an importer it must be prepared to make sure the file
+   * is available.  Preparing the importer will download all the necessary
+   * files.  This value is set to TRUE after the importer is prepared for
+   * funning.
+   */
+  protected $is_prepared;
+
+
+  /**
+   * Stores the last percentage that progress was reported.
+   *
+   * @var integer
+   */
+  protected $reported = 0;
+
+  // --------------------------------------------------------------------------
+  //                          CONSTRUCTORS
+  // --------------------------------------------------------------------------
+  /**
+   * Instantiates a new TripalImporter object.
+   *
+   * @param TripalJob $job
+   *   An optional TripalJob object that this loader is associated with.
+   */
+  public function __construct(Drupal\tripal\Services\TripalJob $job = NULL) {
+
+    // Intialize the private member variables.
+    $this->is_prepared = FALSE;
+    $this->import_id = NULL;
+    $this->arguments = [];
+    $this->job = NULL;
+    $this->total_items = 0;
+    $this->interval = 1;
+    $this->num_handled = 0;
+    $this->prev_update = 0;
+
+
+    // Initialize the logger.
+    $this->logger = \Drupal::service('tripal.logger');  
+
+    // Initialize messenger
+    $this->messenger = \Drupal::messenger();
+  }
+
+  /**
+   * Instantiates a new TripalImporter object using the import record ID.
+   *
+   * This function will automatically instantiate the correct TripalImporter
+   * child class that is appropriate for the provided ID.
+   *
+   * @param $import_id
+   *   The ID of the import recrod.
+   *
+   * @return
+   *   An TripalImporter object of the appropriate child class.
+   */
+  static public function byID($import_id) {
+    // Get the importer.
+    $import = db_select('tripal_import', 'ti')

```

Comment:
We no longer support Drupal 8.9.x. db_select and all other db_* procedural calls have been deprecated for 9.0.x and should be replaced. The change record for this with instructions to update can be found here: https://www.drupal.org/node/2993033